### PR TITLE
Redux-reducers and actions

### PR DESCRIPTION
**Pull request by prius2055**
### Redux - reducers &  actions

**In this branch, I have created the following:**

- categoriesSlice
- bookSlice
- I have also structured my component tree using best practices
- I have also displayed "under construction" for the categories component

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const bookSlice = createSlice({
+  name: 'books',
+  initialState: [],
+  reducers: {
+    addBook() {},
+    removeBook() {},
```
Comment (joel-kalema) [inline src/redux/books/booksSlice.js]: - [ ] Good job working on the "Bookstore: add reducers and actions" project, however, currently we are not able to `add` and `remove` boob. Kindly make sure you add a reducer that adds and a reducer that removes a book as it is [required](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/add_reducers_an_actions_v2.md#:~:text=Includes%20a%20reducer,removes%20a%20book) 😃

**Review by joel-kalema — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @prius2055 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me @joel-kalema  when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/bookItems.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const bookItems = [];
+
+//   {
+//     id: '',
+//     genre: '',
+//     title: '',
+//     author: '',
+//     percentageCompletion: '',
+//     completed: '',
+//   },
```
Comment (Zuhaib042) [inline src/bookItems.js:10]: - [OPTIONAL] I will highly recommend you to remove this commented code as it makes our codebase looks ugly and unprofessional.

**Review by Zuhaib042 — APPROVED** [review]
Hi @prius2055 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ Implemented all suggested changes
✅ Linters passing
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**